scalafix-core/src/main/scala/scalafix/Scalafix.scala

```diff
@@ -1,50 +1,23 @@
 package scalafix
 
-import scala.collection.immutable.Seq
 import scala.meta._
-import scala.meta.inputs.Input
-import scala.meta.parsers.Parsed
-import scalafix.config.ScalafixConfig
-import scalafix.rewrite.RewriteCtx
-import scalafix.rewrite.ScalafixCtx
-import scalafix.rewrite.ScalafixMirror
-import scalafix.util.CanOrganizeImports
-import scalafix.util.Patch
+import scala.util.control.NonFatal
 
 object Scalafix {
-  def fix(code: Input,
-          config: ScalafixConfig,
-          semanticApi: Option[ScalafixMirror]): Fixed = {
-    config.parser.apply(code, config.dialect) match {
-      case Parsed.Success(ast) =>
-        fix(ast, config, semanticApi)
-      case Parsed.Error(pos, msg, e) =>
-        Fixed.Failed(Failure.ParseError(pos, msg, e))
+  def fix(input: Input, config: ScalafixConfig): Fixed =
+    config.dialect(input).parse[Source] match {
+      case parsers.Parsed.Success(ast) =>
+        fix(RewriteCtx.syntactic(ast, config))
+      case parsers.Parsed.Error(pos, msg, details) =>
+        Fixed.Failed(Failure.ParseError(pos, msg, details))
     }
-  }
-  def fix(ast: Tree,
-          config: ScalafixConfig,
-          semanticApi: Option[ScalafixMirror]): Fixed = {
-    implicit val ctx: ScalafixCtx = RewriteCtx(
-      ast,
-      config,
-      semanticApi.getOrElse(ScalafixMirror.failingMirror(config.dialect)))
+
+  def fix(ctx: RewriteCtx): Fixed =
```

Comment:
I'm not really sure what to do with this method. I like having one single entry point for Scalafix, but at the point where you have a RewriteCtx you have already done all of the work required to run the fix. Everything this method does is handle the wrappedRewrite call which is an internal implementation detail, which may change in the future.